docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -10,6 +10,88 @@ which conforms to the schema defined below. You may find it easier to
 start writing your ``DOCUMENTATION`` string in an editor with YAML
 syntax highlighting before you include it in your Python file.
 
+All modules must have the following sections defined in this order:
+
+1. ANSIBLE_METADATA
+2. DOCUMENTATION
+3. EXAMPLES
+4. RETURNS
+5. Python imports
+
+.. note:: Why don't the imports go first?
+
+  Keen Python programmers may notice that contary to PEP8's advice we don't put ``imports`` at the top of the file. This is due to how modules used to be combined with libraries. For more information see :doc:`developing_program_flow_modules`.
+
```

Comment:
The rationale here should be:

This is because the ANSIBLE_METADATA through RETURNS sections are not used by the module code itself. They are essentially extra docstrings for the file.  The imports are placed after these special variables for the same reason as PEP8 puts the imports after the introductory comments and docstrings: It keeps the active parts of the code together and the pieces which are purely informational apart.